src/extent.c

```diff
@@ -929,7 +929,7 @@ extent_record(tsdn_t *tsdn, pac_t *pac, ehooks_t *ehooks, ecache_t *ecache,
 	}
 	if (!ecache->delay_coalesce) {
 		edata = extent_try_coalesce(tsdn, pac, ehooks, ecache, edata,
-		    NULL);
+		    &(bool){false});
```

Comment:
Good catch!

nit: let's still add a plain `bool coalesced_unused;` and pass it here. I'm a bit concerned some of the older compilers might not like it this way.

Let's also add an `assert(coalesced != NULL);` here https://github.com/jemalloc/jemalloc/blob/70344a2d38eb71a162ea19d1a4fee8f0d168588b/src/extent.c#L824